Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessor.kt

```diff
@@ -66,35 +69,40 @@ class CoronaTestProcessor @Inject constructor(
         )
     }
 
-    suspend fun pollServer(test: CoronaTest): ServerResponse {
-
-        return try {
-            val response = try {
-                coronaTestService.checkTestResult(test.registrationToken)
-            } catch (e: BadRequestException) {
-                if (test.isOlderThan21Days(timeStamper.nowUTC)) {
-                    Timber.v("HTTP 400 error after 21 days, remapping to PCR_OR_RAT_REDEEMED.")
-                    CoronaTestResultResponse(coronaTestResult = PCR_OR_RAT_REDEEMED)
-                } else {
-                    throw e
-                }
-            }
-
-            val testResult = when (test.type) {
-                RAPID_ANTIGEN -> response.coronaTestResult.toValidatedRaResult()
-                PCR -> response.coronaTestResult.toValidatedPcrResult()
+    suspend fun pollServer(familyTest: FamilyCoronaTest): PollResult = try {
+        val response = try {
+            coronaTestService.checkTestResult(familyTest.registrationToken)
+        } catch (e: BadRequestException) {
+            if (familyTest.isOlderThan21Days(timeStamper.nowUTC)) {
+                Timber.v("HTTP 400 error after 21 days, remapping to PCR_OR_RAT_REDEEMED.")
+                CoronaTestResultResponse(coronaTestResult = PCR_OR_RAT_REDEEMED)
+            } else {
+                throw e
             }
+        }
 
-            ServerResponse.CoronaTestResultUpdate(
-                coronaTestResult = testResult,
-                sampleCollectedAt = response.sampleCollectedAt,
-                labId = response.labId,
-            )
-        } catch (e: Exception) {
-            Timber.e(e, "Failed to poll server for  %s", test)
-            if (e !is CwaWebException) e.report(ExceptionCategory.INTERNAL)
-            return ServerResponse.Error(e)
+        val testResult = when (familyTest.type) {
+            RAPID_ANTIGEN -> response.coronaTestResult.toValidatedRaResult()
+            PCR -> response.coronaTestResult.toValidatedPcrResult()
         }
+
+        val update = CoronaTestUpdate(
+            coronaTestResult = testResult,
+            sampleCollectedAt = response.sampleCollectedAt,
+            labId = response.labId,
+        )
+
+        PollResult.Success(
+            original = familyTest,
+            updated = familyTest.updateFromResponse(update)
```

Comment:
This will overwrite other updates that have been done simultaneously. I have done it with an update function within a transaction on purpose. We are overwriting all fields that may have been changed from elsewhere in the meantime between reading from database and the end of the backend call.

Reply:
Ok, What do you suggest to avoid this ? Also doing them one by one is is not nice , Root cause of these changes that when moving all tests to recycle bin ,sometimes one was left